src/test/java/org/prebid/server/auction/CurrencyConversionServiceTest.java

```diff
@@ -0,0 +1,271 @@
+package org.prebid.server.auction;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import io.vertx.core.Handler;
+import io.vertx.core.Vertx;
+import io.vertx.core.buffer.Buffer;
+import io.vertx.core.http.HttpClient;
+import io.vertx.core.http.HttpClientRequest;
+import io.vertx.core.http.HttpClientResponse;
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.Test;
+import org.mockito.Mock;
+import org.mockito.junit.MockitoJUnit;
+import org.mockito.junit.MockitoRule;
+import org.mockito.stubbing.Answer;
+import org.prebid.server.VertxTest;
+import org.prebid.server.currency.proto.response.CurrencyConversionRates;
+import org.prebid.server.exception.PreBidException;
+
+import java.math.BigDecimal;
+import java.util.HashMap;
+import java.util.Map;
+
+import static java.util.Collections.singletonMap;
+import static org.assertj.core.api.Assertions.*;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.ArgumentMatchers.anyString;
+import static org.mockito.BDDMockito.given;
+import static org.mockito.Mockito.*;
+
+public class CurrencyConversionServiceTest extends VertxTest {
+
+    private static final String USD = "USD";
+    private static final String GBP = "GBP";
+    private static final String EUR = "EUR";
+    private static final String UAH = "UAH";
+    private static final String AUD = "AUD";
+    private static final String URL = "http://currency.prebid.org/latest.json";
+
+    @Rule
+    public final MockitoRule mockitoRule = MockitoJUnit.rule();
+
+    @Mock
+    private HttpClient httpClient;
+    @Mock
+    private Vertx vertx;
+    @Mock
+    private HttpClientRequest httpClientRequest;
+
+    private CurrencyConversionService currencyService;
+
+    @Before
+    public void setUp() throws JsonProcessingException {
+        final Map<String, Map<String, BigDecimal>> currencyRates = new HashMap<>();
+        currencyRates.put(GBP, singletonMap(EUR, BigDecimal.valueOf(1.15)));
+        currencyRates.put(UAH, singletonMap(EUR, BigDecimal.valueOf(1.1565)));
+        givenHttpClientReturnsResponse(httpClient, 200,
+                mapper.writeValueAsString(CurrencyConversionRates.of(null, currencyRates)));
+
+        currencyService = new CurrencyConversionService(URL, 1L, httpClient, vertx);
+    }
+
+    @Test
+    public void creationShouldFailOnNullArguments() {
```

Comment:
We are deviating from testing constructors with null arguments pass as these tests aren't very useful so this test could be safely dropped.